typescript/sdk/src/hook/read.ts

```diff
@@ -0,0 +1,284 @@
+  deriveAggregationConfig(address: Address): Promise<AggregationHookConfig>;
+  deriveIgpConfig(address: Address): Promise<IgpHookConfig>;
+  deriveProtocolFeeConfig(address: Address): Promise<ProtocolFeeHookConfig>;
+  deriveOpStackConfig(address: Address): Promise<OpStackHookConfig>;
+  deriveDomainRoutingConfig(address: Address): Promise<DomainRoutingHookConfig>;
+  deriveFallbackRoutingConfig(
+    address: Address,
+  ): Promise<FallbackRoutingHookConfig>;
+  derivePausableConfig(address: Address): Promise<PausableHookConfig>;
+}
+
+export class EvmHookReader implements HookReader<ProtocolType.Ethereum> {
+  protected readonly provider: providers.Provider;
+
+  constructor(protected readonly multiProvider: MultiProvider, chain: Chains) {
+    this.provider = this.multiProvider.getProvider(chain);
+  }
+
+  public static stringifyConfig(config: HookConfig, space?: number): string {
+    return JSON.stringify(
+      config,
+      (_, value) => {
+        // Check if the value looks like a serialized BigNumber
+        if (
+          typeof value === 'object' &&
+          value !== null &&
+          value.type === 'BigNumber' &&
+          value.hex
+        ) {
+          return BigNumber.from(value.hex).toString();
+        }
+        // Handle bigint values
+        return typeof value === 'bigint' ? value.toString() : value;
+      },
+      space,
+    );
+  }
+
+  async deriveHookConfig(address: Address): Promise<HookConfig> {
+    const hook = IPostDispatchHook__factory.connect(address, this.provider);
+    const onchainHookType: OnchainHookType = await hook.hookType();
+    const hookType = mapOnchainHookToHookType(onchainHookType);
+
+    switch (hookType) {
+      case HookType.MERKLE_TREE:
+        return this.deriveMerkleTreeConfig(address);
+      case HookType.AGGREGATION:
+        return this.deriveAggregationConfig(address);
+      case HookType.INTERCHAIN_GAS_PAYMASTER:
+        return this.deriveIgpConfig(address);
+      case HookType.PROTOCOL_FEE:
+        return this.deriveProtocolFeeConfig(address);
+      case HookType.OP_STACK:
+        return this.deriveOpStackConfig(address);
+      case HookType.ROUTING:
+        return this.deriveDomainRoutingConfig(address);
+      case HookType.FALLBACK_ROUTING:
+        return this.deriveFallbackRoutingConfig(address);
+      case HookType.PAUSABLE:
+        return this.derivePausableConfig(address);
+      default:
+        throw new Error(`Unsupported HookType: ${hookType}`);
+    }
+  }
+
+  async deriveMerkleTreeConfig(_: Address): Promise<MerkleTreeHookConfig> {
+    return {
+      type: HookType.MERKLE_TREE,
+    };
+  }
+
+  async deriveAggregationConfig(
+    address: Address,
+  ): Promise<AggregationHookConfig> {
+    const hook = StaticAggregationHook__factory.connect(address, this.provider);
+    const hooks = await hook.hooks(ethers.constants.AddressZero);
+    const hookConfigs = await Promise.all(
+      hooks.map(this.deriveHookConfig.bind(this)),
```

Comment:
im not totally sure why but iirc `this.fn.bind(this)` is an antipattern
cc @jmrossy

Reply:
can we just `(hook) => this.deriveHookConfig(hook)`

Reply:
I don't mind it but sure thing, I'll update that here and in the chainmetadata to remove the `.bind`